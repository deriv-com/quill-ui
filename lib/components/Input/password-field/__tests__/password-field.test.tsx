import { fireEvent, render, screen } from "@testing-library/react";
import PasswordField, { PasswordFieldProps } from "..";
import React from "react";

describe("PasswordField", () => {
    const setup = (props: Partial<PasswordFieldProps> = {}) => {
        const placeholder = "Password";
        const utils = render(
            <PasswordField {...props} placeholder={placeholder} />,
        );
        const input = utils.getByPlaceholderText(
            placeholder,
        ) as HTMLInputElement;
        return {
            input,
            ...utils,
        };
    };

    it("should match snapshot", () => {
        const { container } = setup({ placeholder: "Enter password" });
        expect(container).toMatchSnapshot();
    });

    it("should render with default props", () => {
        setup();
        const placeholder = "Password";
        expect(screen.getByPlaceholderText(placeholder)).toBeInTheDocument();
    });

    it("should render with a label", () => {
        setup({ label: "password" });
        expect(screen.getByLabelText("password")).toBeInTheDocument();
    });

    it("should call onChange handler when input changes", () => {
        const handleChange = jest.fn();
        const { input } = setup({ onChange: handleChange });
        fireEvent.change(input, { target: { value: "Password1234" } });
        expect(handleChange).toHaveBeenCalledTimes(1);
    });

    it("should display message when provided", () => {
        const { container } = setup({
            message: "Error message",
            status: "error",
        });
        expect(screen.getByText("Error message")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("toggles password visibility icon", () => {
        const { getByTestId } = render(<PasswordField />);

        // Check that the eye slash icon is initially rendered
        expect(getByTestId("eye-slash-icon")).toBeInTheDocument();

        // Click the eye slash icon to show password
        fireEvent.click(getByTestId("eye-slash-icon"));

        // Check that the eye icon is rendered after clicking
        expect(getByTestId("eye-icon")).toBeInTheDocument();

        // Click the eye icon to hide password
        fireEvent.click(getByTestId("eye-icon"));

        // Check that the eye slash icon is rendered again after clicking
        expect(getByTestId("eye-slash-icon")).toBeInTheDocument();
    });
});
