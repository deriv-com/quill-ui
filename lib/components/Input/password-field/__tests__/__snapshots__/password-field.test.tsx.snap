// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordField should display message when provided 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--error quill-input__wrapper__size--lg password_field"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-no-label__wrapper"
        >
          <input
            class="input peer input__align--left input__size--lg"
            id="quill-input-5"
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <span
          class="icon_wrapper status-icon--error"
        >
          <svg
            height="24"
            role="img"
            viewBox="0 0 32 32"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.688 9.82 7.953 22.516c-.078.117-.078.195-.078.312 0 .313.234.547.547.547h15.117c.313 0 .586-.234.586-.547 0-.117-.04-.195-.117-.312L16.273 9.82A.29.29 0 0 0 16 9.625c-.156 0-.234.078-.312.195m-1.602-.976A2.25 2.25 0 0 1 16 7.75c.742 0 1.484.43 1.875 1.094l7.734 12.695c.235.39.391.82.391 1.29 0 1.327-1.094 2.421-2.46 2.421H8.421A2.434 2.434 0 0 1 6 22.828q0-.703.352-1.289zm3.164 12.031c0 .703-.586 1.25-1.25 1.25-.703 0-1.25-.547-1.25-1.25 0-.664.547-1.25 1.25-1.25.664 0 1.25.586 1.25 1.25m-.312-7.187v3.75c0 .546-.43.937-.938.937-.547 0-.937-.39-.937-.937v-3.75c0-.508.39-.938.937-.938a.95.95 0 0 1 .938.938"
            />
          </svg>
        </span>
        <span
          class="icon_wrapper"
        >
          <svg
            class="password_field__eye-icon"
            data-testid="eye-icon"
            height="24"
            role="img"
            viewBox="0 0 32 32"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.32 11.813C7.64 13.375 6.508 15.25 6 16.5c.508 1.25 1.64 3.125 3.32 4.688S13.227 24 16 24c2.734 0 4.96-1.25 6.64-2.812 1.68-1.563 2.813-3.438 3.36-4.688-.547-1.25-1.68-3.125-3.36-4.687S18.735 9 16 9c-2.773 0-5 1.25-6.68 2.813M16 7.75c3.125 0 5.664 1.445 7.5 3.164 1.836 1.68 3.047 3.711 3.633 5.117a1.4 1.4 0 0 1 0 .977c-.586 1.367-1.797 3.398-3.633 5.117S19.125 25.25 16 25.25c-3.164 0-5.703-1.406-7.54-3.125-1.835-1.719-3.046-3.75-3.632-5.117a1.4 1.4 0 0 1 0-.977c.586-1.406 1.797-3.437 3.633-5.117C10.297 9.195 12.836 7.75 16 7.75m-3.75 8.75c0 1.367.703 2.578 1.875 3.281 1.133.664 2.578.664 3.75 0 1.133-.703 1.875-1.914 1.875-3.281 0-1.328-.742-2.54-1.875-3.242-1.172-.664-2.617-.664-3.75 0-1.172.703-1.875 1.914-1.875 3.242m8.75 0c0 1.797-.977 3.438-2.5 4.336-1.562.898-3.477.898-5 0A4.97 4.97 0 0 1 11 16.5c0-1.758.938-3.398 2.5-4.297 1.523-.898 3.438-.898 5 0 1.523.899 2.5 2.54 2.5 4.297"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="message__container message__container--lg message__container__status--error"
    >
      <span
        class="message__container__text"
      >
        Error message
      </span>
    </div>
  </div>
</div>
`;

exports[`PasswordField should match snapshot 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--lg password_field"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-no-label__wrapper"
        >
          <input
            class="input peer input__align--left input__size--lg"
            id="quill-input-1"
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <span
          class="icon_wrapper"
        >
          <svg
            class="password_field__eye-icon"
            data-testid="eye-icon"
            height="24"
            role="img"
            viewBox="0 0 32 32"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.32 11.813C7.64 13.375 6.508 15.25 6 16.5c.508 1.25 1.64 3.125 3.32 4.688S13.227 24 16 24c2.734 0 4.96-1.25 6.64-2.812 1.68-1.563 2.813-3.438 3.36-4.688-.547-1.25-1.68-3.125-3.36-4.687S18.735 9 16 9c-2.773 0-5 1.25-6.68 2.813M16 7.75c3.125 0 5.664 1.445 7.5 3.164 1.836 1.68 3.047 3.711 3.633 5.117a1.4 1.4 0 0 1 0 .977c-.586 1.367-1.797 3.398-3.633 5.117S19.125 25.25 16 25.25c-3.164 0-5.703-1.406-7.54-3.125-1.835-1.719-3.046-3.75-3.632-5.117a1.4 1.4 0 0 1 0-.977c.586-1.406 1.797-3.437 3.633-5.117C10.297 9.195 12.836 7.75 16 7.75m-3.75 8.75c0 1.367.703 2.578 1.875 3.281 1.133.664 2.578.664 3.75 0 1.133-.703 1.875-1.914 1.875-3.281 0-1.328-.742-2.54-1.875-3.242-1.172-.664-2.617-.664-3.75 0-1.172.703-1.875 1.914-1.875 3.242m8.75 0c0 1.797-.977 3.438-2.5 4.336-1.562.898-3.477.898-5 0A4.97 4.97 0 0 1 11 16.5c0-1.758.938-3.398 2.5-4.297 1.523-.898 3.438-.898 5 0 1.523.899 2.5 2.54 2.5 4.297"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
